services/QuillLMS/app/services/student_learning_sequences/fill_missing/post_diagnostics.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module StudentLearningSequences
+  module FillMissing
+    class PostDiagnostics < Base
+      def sql
+        <<-SQL
+          SELECT
+                  classroom_units.id AS classroom_unit_id, assigned AS student_id
+              FROM classroom_units
+              CROSS JOIN UNNEST(classroom_units.assigned_student_ids) AS assigned
+              JOIN units ON classroom_units.unit_id = units.id
+              JOIN unit_activities ON classroom_units.unit_id = unit_activities.unit_id
+              LEFT OUTER JOIN student_learning_sequences
+                  ON classroom_units.id = student_learning_sequences.initial_classroom_unit_id
+                      AND student_learning_sequences.user_id = assigned
+              LEFT OUTER JOIN student_learning_sequence_activities
+                  ON student_learning_sequences.id = student_learning_sequence_activities.student_learning_sequence_id
+                      AND student_learning_sequences.initial_classroom_unit_id = student_learning_sequence_activities.classroom_unit_id
+                      AND student_learning_sequences.initial_activity_id = student_learning_sequence_activities.activity_id
+              WHERE unit_activities.activity_id IN (1774, 1814, 1818, 1664, 1669, 1680, 2538, 2540, 2542, 2551, 2557, 2564)
```

Comment:
Let's make the activity list a constant (we recently finished cleaning up the older references in the codebase last year)

Reply:
Good call.  I'll update.